File: 프로그래머스/unrated/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    answer = []
    mj=[]
    for s in score:
        mj.append(s)
        mj.sort(reverse=True)
        if len(mj)<k : answer.append(mj[-1])
        else : answer.append(mj[k-1])
        
    return answer

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.java
import java.util.*;
class Solution {
    public int[] solution(int k, int[] score) {
        int[] answer = new int[score.length];
        List<Integer> list=new ArrayList<>();
        if(score.length<k) k=score.length;
        for(int i=0;i<k;i++){
        
            list.add(score[i]);
            Collections.sort(list); 
            answer[i]=list.get(0);
        }
       
        for(int i=k;i<score.length;i++){
            list.add(score[i]);
            Collections.sort(list);
            answer[i]=list.get(list.size()-k);
            
        }
        return answer;
    }
}